ports/psoc6/mpconfigport.h

```diff
@@ -86,139 +84,66 @@
-#define MICROPY_PY_MACHINE_SOFTSPI              (0)
-#define MICROPY_PY_ONEWIRE                      (0)
+// VFS
 #define MICROPY_VFS                             (1)
-
-#define MICROPY_VFS_FAT                         (0)
+#define MICROPY_READER_VFS                      (1)
 
 // fatfs configuration
 #define MICROPY_FATFS_ENABLE_LFN                (1)
 #define MICROPY_FATFS_LFN_CODE_PAGE             437 /* 1=SFN/ANSI 437=LFN/U.S.(OEM) */
 #define MICROPY_FATFS_RPATH                     (2)
 
-// set to 1 to enable filesystem to be loaded on external qspi flash
-// if set to 0, filesystem is located in an allotted area of internal flash of PSoC6
-#define MICROPY_ENABLE_EXT_QSPI_FLASH               (1)
-
-#ifndef MICROPY_BOARD_ENTER_BOOTLOADER
-#define MICROPY_BOARD_ENTER_BOOTLOADER(nargs, args)
-#endif
-
 // By default networking should include sockets, ssl, websockets, webrepl, dupterm.
-// #define MICROPY_PY_NETWORK              (1)
-
-#if MICROPY_PY_NETWORK
-
-#ifndef MICROPY_PY_USOCKET
-#define MICROPY_PY_USOCKET              (1)
-#endif
-
-#ifndef MICROPY_PY_USSL
-#define MICROPY_PY_USSL                 (1)
-#endif
-
-#ifndef MICROPY_PY_UWEBSOCKET
-#define MICROPY_PY_UWEBSOCKET           (1)
-#endif
-
-#ifndef MICROPY_PY_UHASHLIB_SHA1
-#define MICROPY_PY_UHASHLIB_SHA1        (1)
-#endif
+// #define MICROPY_PY_NETWORK                      (1)
+// #define MICROPY_PY_USOCKET                      (1)
+// #define MICROPY_PY_UWEBSOCKET                   (1)
+// #define MICROPY_PY_WEBREPL                      (1)
+#define MICROPY_PY_OS_DUPTERM                      (1)
 
-#ifndef MICROPY_PY_WEBREPL
-#define MICROPY_PY_WEBREPL              (1)
-#endif
+#define MP_STATE_PORT                           MP_STATE_VM
 
-#ifndef MICROPY_PY_OS_DUPTERM
-#define MICROPY_PY_OS_DUPTERM           (1)
-#endif
+#define MICROPY_PY_UCRYPTOLIB                   (1)
+#define MICROPY_PY_UCRYPTOLIB_CTR               (1)
+#define MICROPY_PY_UCRYPTOLIB_CONSTS            (1)
 
-#endif
-
-
-#define MICROPY_TRACKED_ALLOC        (MICROPY_SSL_MBEDTLS)
-
-#define MICROPY_PY_UCRYPTOLIB        (1)
-#define MICROPY_PY_UCRYPTOLIB_CTR    (1)
-#define MICROPY_PY_UCRYPTOLIB_CONSTS (1)
-
-#define MICROPY_PY_UHASHLIB          (1)
-#define MICROPY_PY_UHASHLIB_MD5      (1)
-#define MICROPY_PY_UHASHLIB_SHA1     (1)
-#define MICROPY_PY_UHASHLIB_SHA256   (1)
-
-#define MP_STATE_PORT MP_STATE_VM
+#define MICROPY_PY_UHASHLIB_MD5                 (1)
+#define MICROPY_PY_UHASHLIB_SHA1                (1)
 
 // Miscellaneous settings
-#define MICROPY_MAKE_POINTER_CALLABLE(p) ((void *)((mp_uint_t)(p) | 1))
+#define MICROPY_MAKE_POINTER_CALLABLE(p)        ((void *)((mp_uint_t)(p) | 1))
 
 #define MP_SSIZE_MAX (0x7fffffff)
-typedef intptr_t mp_int_t; // must be pointer size
+typedef intptr_t mp_int_t;  // must be pointer size
 typedef uintptr_t mp_uint_t; // must be pointer size
 typedef intptr_t mp_off_t;
 
 // We need to provide a declaration/definition of alloca()
 #include <alloca.h>
```

Comment:
Any special reason for including this header? Why not at the top with stdint?

Reply:
Good point. Put here in order to be close to the code that makes use of it, but maybe better at the top => done.